pkg/goDB/DBWorkManager.go

```diff
@@ -163,76 +225,265 @@ func (w *DBWorkManager) CreateWorkerJobs(tfirst int64, tlast int64, query *Query
+
+			aggMetadata, err = w.readMetadataAndEvaluate(curDir, query,
+				blocks, offset,
+				aggMetadata, func(metadata *InterfaceMetadata, stats gpfile.Stats) gpfile.Stats {
+					return metadata.Stats.Sub(stats)
+				},
+			)
+			if err != nil {
+				return nil, fmt.Errorf("failed to read block metadata: %w", err)
+			}
+			w.tLastCovered = curDir.BlockMetadata[0].BlockList[tLastBlockInd].Timestamp
+		} else {
+			logger.Debug("setting last covered timestamp to last")
 			w.tLastCovered = dirLast
 		}
+
+		logger.Debug("closing last directory")
+
+		// TODO: the tool will just get stuck here if the tfirst is in the last available directory
+		// for that interface
+		//
+		//	goquery --logging.level=debug -d /root/db-v4 -f '28.6.23 23:00'  list eth0
+		//
+		// This may be a related phenomenon to the panic observed further below
 		if err := curDir.Close(); err != nil {
-			return false, fmt.Errorf("failed to close last GPDir %s after ascertaining query block timing: %w", curDir.Path(), err)
+			return nil, fmt.Errorf("failed to close last GPDir %s after ascertaining query block timing: %w", curDir.Path(), err)
 		}
+
+		logger.Debug("last directory closed")
 	}
 
-	return 0 < numDirs, nil
+	// assign time range of interface
+	//
+	// IMPORTANT: in the case of the first timestamp, the DB write interval is subtracted to
+	// show that flows are taken into account _up to_ the timestamp of the effective write out.
+	//
+	// This will retain consistency with what is presented in the summary after a normal query
+	aggMetadata.First, aggMetadata.Last = w.GetCoveredTimeInterval()
+
+	return aggMetadata, nil
+}
+
+// NOTE: contrary to it's bigger sister readBlocksAndEvaluate, the function assumes that the workDir is already open.
+// This is owed to the nature of its calling function
+func (w *DBWorkManager) readMetadataAndEvaluate(workDir *gpfile.GPDir, query *Query, blocks []storage.BlockAtTime, offset int, aggMetadata *InterfaceMetadata,
+	statsOpFunc func(*InterfaceMetadata, gpfile.Stats) gpfile.Stats,
+) (*InterfaceMetadata, error) {
+	logger := logging.Logger().With("iface", w.iface, "day", workDir.Path())
+
+	var (
+		bytesRcvdValues, bytesSentValues, pktsRcvdValues, pktsSentValues []uint64
+		err                                                              error
+	)
+
+	// TODO: results in a panic
+	//
+	// panic: runtime error: index out of range [24] with length 0
+	// goroutine 1 [running]:
+	// github.com/els0r/goProbe/pkg/goDB/storage/gpfile.(*GPFile).ReadBlockAtIndex(0x4?, 0xc0000143a8?)
+	//         /usr/src/packages/src/sdwan/security/pkg/OSAGtmi/goProbe/pkg/goDB/storage/gpfile/gpfile.go:136 +0x60d
+	// github.com/els0r/goProbe/pkg/goDB/storage/gpfile.(*GPDir).ReadBlockAtIndex(0xc0000243c0, 0x4, 0x0?)
+	//
+	// 	goquery --logging.level=debug -d /root/db-v4 -f '28.6.23 23:00' -l '28.6.23 23:30:00' list eth0
+	//
+	// 7 blocks to evaluate, starts at 23:00:00, ends at 23:30:00. First relevant index for subtraction is 24
```

Comment:
This is the first bug described in commit. Any idea why the `gpfile` is empty? It shouldn't, since 32 blocks are stored in it and `goquery -i eth0 iface` can deal with the same time frame perfectly.